### PR TITLE
Adds ability to delete block types via the CLI

### PR DESCRIPTION
**Pull request by anticorrelator**
Enables the deletion of Block Types. Also ensures that important "protected" block types cannot be destructively modified. We do this by ensuring that any route that attempts to mutate or delete an existing Block Type updates the protection status of relevant Block Types.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a `fix`, `feature`, `enhancement`, `docs`, `collections`, or `maintenance` label categorizing the change.

### CODE REVIEW

**Comment by desertaxle** [comment]
I think this is good to go once the merge conflicts are resolved.

**Comment by madkinsz** [comment]
Sweet!

Can you update the title as well so it works as an auto-generated changelog entry?

**Review by desertaxle — APPROVED** [review]
LGTM

**Pull request merged**
